app/resources.py

```diff
@@ -0,0 +1,149 @@
+import random
+import hashlib
+from typing import (
+    MutableMapping,
+    Type,
+)
+from flask_restful import (
+    Resource,
+    reqparse,
+    abort,
+)
+from flask_jwt_extended import (
+    create_access_token,
+    create_refresh_token,
+    jwt_required,
+    jwt_refresh_token_required,
+    get_jwt_identity,
+    get_raw_jwt
+)
+from models import User
+from extensions import (
+    jwt,
+    db,
+)
+
+
+@jwt.user_claims_loader
+def add_claims_to_access_token(user: User):
+    return {
+        'username': user.username,
+    }
+
+
+@jwt.user_identity_loader
+def user_identity_lookup(user: User):
+    return user.id
+
+
+class BaseResource(Resource):
+    __resources: MutableMapping[str, Type[Resource]] = {}
+
+    def __init_subclass__(cls, *args, **kwargs):
+        cls.__resources[cls.path] = cls
+        super().__init_subclass__(*args, **kwargs)
+
+    @classmethod
+    def register(cls, api):
+        for path, res in cls.__resources.items():
+            api.add_resource(res, path)
+
+
+class UserRegistration(BaseResource):
+    path = "/v1/register"
+
+    def __init__(self):
+        self.parser = reqparse.RequestParser()
+        self.parser.add_argument(
+            'username', help='This field cannot be blank', required=True)
+        self.parser.add_argument(
+            'password', help='This field cannot be blank', required=True)
+        self.parser.add_argument(
+            'email', help='This field cannot be blank', required=True)
+
+    def post(self):
+        data = self.parser.parse_args()
+        username = data['username']
+        password = data['password']
+        email = data['email']
+        password_salt = ''.join(random.choice('0123456789ABCDEF')
+                                for i in range(16))
+        password_with_salt = password + password_salt
+        password_hash = hashlib.sha256(
```

Comment:
Actually instead of using sha256 and generating salt yourself I would use bcrypt which takes care of that.

Reply:
There is a package already https://pypi.org/project/bcrypt/

Reply:
Ok